### PR TITLE
Pass comma delimited string of vendors with consent to GTM

### PR DESCRIPTION
**Pull request by GHaberis**
## What are you doing in this PR?

This PR updates `googleTagManager.js`, I've created a new string variable `vendorConsentsLookup` in this module. Whenever a user updates their consent via the CMP we set the value of `vendorConsentsLookup` to be a comma delimited string of vendor names (eg. "google-analytics") that have had consent granted to them.

This variable `vendorConsentsLookup` is now pushed to Google Tag Manager (GTM), and will look something like "google-analytics,twitter,bing".

In GTM we can set a conditional check for each tag that must be met for the tag to be enabled. We have the ability to use a regular expression as part of this condition. I therefore propose for each tag we query `vendorConsentsLookup` in GTM using a regular expression. Eg. For Google Analytics to run we check the condition `vendorConsentsLookup.match(new RegExp("(?:^|,)"+"google-analytics"+"(?:,|$)"))`. GTM uses javascript regular expression syntax.

### Next steps

In GTM you can create changes to tag rules which are in a draft state. You can also target Dev/Staging environments in GTM (support.thegulocal.com, support.code.dev-theguardian.com) with the draft rules. I need to do this to confirm the regex lookup is working as expected. Once I've done this I can publish the changes in GTM once this PR has been merged.

[**Trello Card**](https://trello.com/c/NKVX4rAT/3267-tcfv2-google-tag-manager-granular-vendor-controls)

## Why are you doing this?

This will allow us to continue to use GTM (if a user has consented to GTM) but enable/disable tags within it based on a user's consent at a vendor level.

### CODE REVIEW

File: support-frontend/assets/helpers/tracking/googleTagManager.js [hunk 1]
```diff
@@ -277,16 +295,22 @@ function init(participations: Participations) {
...
     [key: string]: boolean
   }) => {
+
+    // Update vendorConsentsLookup value based on thirdPartyTrackingConsent
+    vendorConsentsLookup = Object.keys(thirdPartyTrackingConsent).reduce((accumulator, vendorKey) => {
+      if (thirdPartyTrackingConsent[vendorKey]) {
+        return [...accumulator, vendorKey];
+      }
+      return [...accumulator];
+    }, []).join(',');
```
Comment (paulbrown1982) [inline support-frontend/assets/helpers/tracking/googleTagManager.js:305]: Is [.filter()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) supported? e.g

```
vendorConsentsLookup = Object.keys(thirdPartyTrackingConsent).filter((vendorKey) => thirdPartyTrackingConsent[vendorKey]).join(',')
```

File: support-frontend/assets/helpers/tracking/googleTagManager.js [hunk 2]
```diff
@@ -29,8 +29,9 @@ type GaEventData = {
   label: ?string,
 }
 
-const googleAnalyticsKey = 'googleAnalytics';
-const googleTagManagerKey = 'googleTagManager';
+// these values match the keys used by @guardian/consent-management-platform
```
Comment (paulbrown1982) [inline support-frontend/assets/helpers/tracking/googleTagManager.js:32]: 👍

**Review by paulbrown1982 — APPROVED** [review]
👍  LGTM, just a tiny tweak to how you filter the vender ids if you feel necessary.

File: support-frontend/assets/helpers/tracking/googleTagManager.js (hunk 1 above)
Comment (GHaberis) [reply support-frontend/assets/helpers/tracking/googleTagManager.js:305]: Yes it is, this is a nice suggestion, happy to adopt this 👍

**Comment by GHaberis** [comment]
@paulbrown1982 I've temporarily updated the assignment of `userConsentsToGTM` to check GTM *and GA*, rather than GTM alone. This matches the logic currently in production. I've done this is it will allow me to publish this PR in advance to publishing my Google Tag Managers configuration update. Next steps in order:

1) Release this PR
2) Publish Google Tag Managers configuration update that checks `vendorConsentsLookup` for GA.
3) Release PR that removes GA check from `userConsentsToGTM` assignment
4) Release PR that adds all additional vendors in GTM to the `vendors` list, and therefore the `vendorConsentsLookup`
5) Update Google Tag Managers configuration to check `vendorConsentsLookup` for each tag/vendor.

I've prepared and tested the Google Tag Manager configuration update in stage 2 already and it works as expected, only enabling GA if "google-analytics" is matched in the `vendorConsentsLookup` comma delimited list.

**Review by paulbrown1982 — APPROVED** [review]
👍  Sounds like a good plan.

**Pull request merged**